nomos-services/api/src/http/storage/adapters/rocksdb.rs

```diff
@@ -0,0 +1,297 @@
+use std::{
+    collections::HashSet,
+    fmt::{Debug, Display},
+    hash::Hash,
+    io::ErrorKind,
+    marker::PhantomData,
+};
+
+use bytes::Bytes;
+use futures::{stream, Stream, StreamExt as _};
+use nomos_core::{
+    block::Block,
+    da::blob::{LightShare, Share},
+    header::HeaderId,
+};
+use nomos_storage::{
+    api::da::StorageDaApi,
+    backends::{rocksdb::RocksBackend, StorageSerde},
+    StorageMsg, StorageService,
+};
+use overwatch::{
+    services::{relay::OutboundRelay, AsServiceId, ServiceData},
+    DynError,
+};
+use serde::{de::DeserializeOwned, Serialize};
+use tokio::sync::oneshot;
+
+use crate::{http::storage::StorageAdapter, wait_with_timeout};
+
+pub struct RocksAdapter<StorageOp, RuntimeServiceId>
+where
+    StorageOp: StorageSerde + Send + Sync + 'static,
+    RuntimeServiceId: Debug + Sync + Display,
+{
+    pub storage_relay: OutboundRelay<
```

Comment:
In the http backend the `HttpStorageAdapter` is used without initialization, could this be removed? Also if it's not used, the `new` method could also be removed.